### PR TITLE
state_title_screen & camera funcs

### PR DESCRIPTION
**Pull request by ethteck**
as well as a fix for pycparser when PERMUTER isn't defined

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀 This PR matches 5 functions (+0.09%) on `us`.

✅ This PR fixes 15 warnings!

File: include/script_api/macros.h [hunk 1]
```diff
@@ -130,10 +130,12 @@
 /// This macro expands to the given opcode and argv, with argc calculated automatically.
 
 #ifndef PERMUTER
+#ifndef M2CTX
 #define EVT_CMD(opcode, argv...) \
     opcode, \
     (sizeof((Bytecode[]){argv})/sizeof(Bytecode)), \
     ##argv
+#endif
```
Comment (nanaian) [inline include/script_api/macros.h:138]:
```suggestion
#ifndef M2CTX
#define EVT_CMD(opcode, argv...) \
    opcode, \
    (sizeof((Bytecode[]){argv})/sizeof(Bytecode)), \
    ##argv
#else
#define EVT_CMD(opcode, argv...)
#endif
```

File: src/state_title_screen.c [hunk 2]
```diff
@@ -42,14 +62,240 @@ Gfx D_80077A50[] = {
...
+            clear_hud_element_cache();
+            spr_init_sprites(0);
+            clear_entity_data(1);
+            clear_windows();
+            gOverrideFlags &= ~GLOBAL_OVERRIDES_8;
+            gOverrideFlags &= ~GLOBAL_OVERRIDES_DISABLE_RENDER_WORLD;
+            temp = D_800779C0;
+            gGameStatusPtr->entryID = 0;
+
+            switch (temp[0]) {
```
Comment (nanaian) [inline src/state_title_screen.c:276]: Is this temp required?

File: src/state_title_screen.c [hunk 3]
```diff
@@ -42,14 +62,240 @@ Gfx D_80077A50[] = {
...
+                case 9:
+                    gGameStatusPtr->creditsViewportMode = 0;
+                    set_game_mode(0x10);
```
Comment (nanaian) [inline src/state_title_screen.c:279]: enum

**Review by nanaian — CHANGES_REQUESTED** [review]
Can you add an enum for `gGameStatusPtr->introState` with values like INTRO_STATE_0, INTRO_STATE_1, etc so its easier to name them later :pray:

**Pull request merged**